lib/iris/_concatenate.py

```diff
@@ -991,6 +1019,14 @@ def register(
             if existing_order == _CONSTANT and this_order != _CONSTANT:
                 self._coord_signature.dim_order[dim_ind] = this_order
 
+        if mismatch_error_msg:
+            if not match and error_on_mismatch:
+                raise iris.exceptions.ConcatenateError([mismatch_error_msg])
+            elif not match:
```

Comment:
```suggestion
        if mismatch_error_msg and not match:
            if error_on_mismatch:
                raise iris.exceptions.ConcatenateError([mismatch_error_msg])
            else:
```

I _think_ this is the same?

Neat structure, by the way 🎖